### PR TITLE
DEX-1307 Automatically set version number in setup.py

### PR DESCRIPTION
**Pull request by karenc**
Use `git describe --tag` to get the tag.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -30,11 +30,28 @@ def _minimal_ext_cmd(cmd):
...
+
+def git_revision():
+    version = git_cmd(['git', 'rev-parse', 'HEAD'])
+    if not version:
+        version = 'unknown-git'
+    return version
+
+
+def git_version():
+    version = git_cmd(['git', 'describe', '--tag'])
```
Comment (bluemellophone) [inline setup.py:47]: Running this code locally results in an `app/version.py` file:

```python
# THIS FILE IS GENERATED FROM SETUP.PY
version = 'v1.0.2.ce9aa231'
git_revision = 'ce9aa231a2c12df2a61b0569d628a09e480a344b'
full_version = '%(version)s.%(git_revision)s' % {
    'version': version,
    'git_revision': git_revision,
}
```

Let's strip the `v` prefix when loading the version from the code.  Ideally, we could also do the standard version output:

```python
import app
print(app.__version__)
```

to mirror the standard example like this:

```python
In [7]: import flask

In [8]: flask.__version__
Out[8]: '1.1.2'
```

**Review by bluemellophone — CHANGES_REQUESTED** [review]
LGTM, minus a small formatting and standardization comment

File: setup.py (hunk 1 above)
Comment (karenc) [reply setup.py:47]: Yeah that's a good idea.  I'll change it

File: tests/test_setup.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        import setup
+
+        importlib.reload(setup)
+
+        setup.write_version_py(filename=str(version_py))
+        module = importlib.import_module(version_py.stem)
+        assert module.version == version
+        assert module.git_revision == git_revision
+        assert module.full_version == f'{version}.{git_revision}'
```
Comment (karenc) [inline tests/test_setup.py:60]: I noticed that the full version for a non tag version looks like `v1.0.2.6ae69c6f.6ae69c6f05b78a1da5616179605ffec22b7fef89`, so the hash is repeated.  But then I don't see any code using this `full_version`, so should I fix it?

File: setup.py (hunk 1 above)
Comment (karenc) [reply setup.py:47]: I added `__version__` to both the flask app and the python `app` package:

```
In [1]: app
Out[1]: <Flask 'app'>

In [2]: app.__version__
Out[2]: '1.0.2.6a304eef'

In [3]: from app import __version__

In [4]: __version__
Out[4]: '1.0.2.6a304eef'
```

File: tests/test_setup.py (hunk 2 above)
Comment (bluemellophone) [reply tests/test_setup.py:60]: honestly, the full version can likely go away

Comment (karenc) [reply tests/test_setup.py:60]: I'll remove it

**Review by bluemellophone — APPROVED** [review]
LGTM

**Review by naknomum — APPROVED** [review]
looks good.

**Pull request merged**